test/spec/modules/admediaBidAdapter_spec.js

```diff
@@ -0,0 +1,256 @@
+    });
+
+    it('should contain valid query string params', () => {
+      sandbox.stub(adLoader, 'loadScript');
+      adapter.callBids(request_single_adunit);
+      expect(adLoader.loadScript.firstCall.args[0]).to.include('//b.admedia.com/banner/prebid/bidder/');
+      expect(adLoader.loadScript.firstCall.args[0]).to.include('aid=1234');
+      expect(adLoader.loadScript.firstCall.args[0]).to.include('&size=300x250');
+      expect(adLoader.loadScript.firstCall.args[0]).to.include('&promo_sizes=300x600');
+      expect(adLoader.loadScript.firstCall.args[0]).to.include('siteDomain');
+      expect(adLoader.loadScript.firstCall.args[0]).to.include('sitePage');
+      expect(adLoader.loadScript.firstCall.args[0]).to.include('callbackId');
+    });
+  });
+
+  describe('Bid Response Validation', () => {
+    let bidObject;
+    let placementCode;
+
+    beforeEach(() => {
+      adapter = new Adapter();
+      sandbox.stub(bidManager, 'addBidResponse');
+
+      bidsRequestedOriginal = $$PREBID_GLOBAL$$._bidsRequested;
+      $$PREBID_GLOBAL$$._bidsRequested = [];
+
+      $$PREBID_GLOBAL$$._bidsRequested.push(request_single_adunit);
+
+      let bidderReponse = {
+        'cpm': 0.92,
+        'ad': 'Banner Ad',
+        'width': '300',
+        'height': '250',
+        'callback_id': '22370f7a5e4575'
+      };
+
+      $$PREBID_GLOBAL$$.admediaHandler(bidderReponse);
+
+      placementCode = bidManager.addBidResponse.firstCall.args[0];
+      bidObject = bidManager.addBidResponse.firstCall.args[1];
+    });
+
+    afterEach(() => {
+      $$PREBID_GLOBAL$$._bidsRequested = bidsRequestedOriginal;
+    });
+
+    it('should add a bid object for each response', () => {
+      sinon.assert.calledOnce(bidManager.addBidResponse);
+    });
+
+    it('should pass the correct placement code as first param', () => {
+      expect(placementCode).to.eql('div-gpt-ad-1');
+    });
+
+    it('should have a good statusCode', () => {
+      expect(bidObject.getStatusCode()).to.eql(CONSTANTS.STATUS.GOOD);
+    });
+
+    it('should add the CPM to the bid object', () => {
+      expect(bidObject).to.have.property('cpm', 0.92);
+    });
+
+    it('should include the ad to the bid object', () => {
+      expect(bidObject).to.have.property('ad');
+    });
+
+    it('should include the size to the bid object', () => {
+      expect(bidObject).to.have.property('width', '300');
+      expect(bidObject).to.have.property('height', '250');
+    });
+
+    it('should include bidder code admedia', () => {
+      expect(bidObject).to.have.property('bidderCode', 'admedia');
+    });
+  });
+
+  describe('usersyncs', () => {
+    beforeEach(() => {
+      adapter = new Adapter();
+      adapter.callBids(request_single_adunit);
+      sandbox.stub(userSync, 'triggerUserSyncs');
+    });
+
+    afterEach(() => {
+    });
+
+    it('should call usersync after default 3000 seconds', () => {
+      window.setTimeout(function() {
```

Comment:
A couple of problems with the setTimeout here.  
1.  you're not using an async callback (usually called `done`) to notify the framework the test is finished.
2. if this works, its adding a 3 second delay every time the tests are run.

I think a better approach would be to test that your `usersync.registerSync()` calls are happening as expected and let the `usersync` module tests validate that they are triggered properly.

Reply:
Agreed. Updated the tests.